### ISSUE TITLE
Che keeps failing on minishift

### ISSUE THREAD

**Issue by tsmaeder** [eclipse-che/che#13995]
### Describe the bug
The che pod keeps failing for me: I'm following the process described in https://github.com/eclipse/che-theia/blob/master/CONTRIBUTING.md

What it looks like is like somehow, a _lot_ of che pods are started (see screenshots). After 5 mins or so, they all fail. 

## Che version

 - [ ] latest
 - [x] nightly
 - [ ] other: please specify

### Steps to reproduce
minishift delete
minishift start --disk-space 40G --memory 16G
oc login -u system:admin
chectl server:start -p minishift -a operator
follow the "contributing" process for che-theia.

### Runtime
   - [ ] kubernetes (include output of `kubectl version`) 
   - [ ] Openshift (include output of `oc version`)
   - [ ] minikube (include output of `minikube version` and `kubectl version`)
   - [x] minishift (include output of `minishift version` and `oc version`)
   - [ ] docker-desktop + K8S (include output of `docker version` and `kubectl version`)
   - [ ] other: (please specify)

### Screenshots
![PodsEvicted](https://user-images.githubusercontent.com/13163770/61791939-b8d7a100-ae1b-11e9-9c69-2e3c8135bda0.png)
![manypods](https://user-images.githubusercontent.com/13163770/61791941-bb39fb00-ae1b-11e9-9828-886bcdbb682b.png)

### Installation method
   - [x] chectl 
   - [x] che-operator 
   - [ ] minishift-addon
   - [ ] I don't know

### Environment
   - [ ] my computer 
       - [ ] Windows
       - [x] Linux
       - [ ] macOS
   - [ ] Cloud
       - [ ] Amazon
       - [ ] Azure
       - [ ] GCE
       - [ ] other (please specify) 
   - [ ] other: please specify

**Comment by tsmaeder** [comment]
This is a blocker for me: I'm unable to progress on my task for 7.0 (Java LS "Classpath is incomplete" warning when loading petclinic #13427)

**Comment by benoitf** [comment]
Could you add logs of che-server pod ? out-of-memory ?

and could you list all kubernetes/openshift deployments objects

**Comment by benoitf** [comment]
and the log of the operator (as you're using operator deployment)
+ checluster objects

**Comment by tsmaeder** [comment]
operator log:
```
﻿time="2019-07-24T11:32:39Z" level=info msg="Go Version: go1.11.5" 
time="2019-07-24T11:32:39Z" level=info msg="Go OS/Arch: linux/amd64" 
time="2019-07-24T11:32:39Z" level=info msg="operator-sdk Version: v0.5.0" 
time="2019-07-24T11:32:39Z" level=info msg="Operator is running on OpenShift v3.x" 
time="2019-07-24T11:32:49Z" level=info msg="Registering Components" 
time="2019-07-24T11:32:49Z" level=info msg="Starting the Cmd" 
time="2019-07-24T11:33:02Z" level=info msg="Waiting for deployment keycloak. Default timeout: 420 seconds" 
time="2019-07-24T11:35:05Z" level=info msg="Deployment 'che' successfully scaled to 1" 
time="2019-07-24T11:35:05Z" level=info msg="Updating eclipse-che CR with Keycloak URL status: http://keycloak-che.192.168.99.114.nip.io " 
time="2019-07-24T11:35:05Z" level=info msg="Custom resource eclipse-che updated" 
time="2019-07-24T11:35:05Z" level=info msg="Updating eclipse-che CR with status: Eclipse Che server: Available" 
time="2019-07-24T11:35:05Z" level=info msg="Custom resource eclipse-che updated" 
time="2019-07-24T11:35:05Z" level=info msg="Updating eclipse-che CR with Eclipse Che server URL: http://che-che.192.168.99.114.nip.io " 
time="2019-07-24T11:35:05Z" level=info msg="Custom resource eclipse-che updated" 
time="2019-07-24T11:35:05Z" level=info msg="Eclipse Che is now available at: http://che-che.192.168.99.114.nip.io " 
time="2019-07-24T11:42:26Z" level=info msg="Updating eclipse-che CR with status: Che API: Unavailable" 
time="2019-07-24T11:42:26Z" level=info msg="Custom resource eclipse-che updated" 
time="2019-07-24T11:42:26Z" level=info msg="Waiting for deployment che. Default timeout: 420 seconds" 
time="2019-07-24T11:48:13Z" level=info msg="Deployment 'che' successfully scaled to 1" 
time="2019-07-24T11:48:13Z" level=info msg="Updating eclipse-che CR with Keycloak URL status: http://keycloak-che.192.168.99.114.nip.io " 
time="2019-07-24T11:48:13Z" level=info msg="Custom resource eclipse-che updated" 
time="2019-07-24T11:48:13Z" level=info msg="Updating eclipse-che CR with status: Eclipse Che server: Available" 
time="2019-07-24T11:48:13Z" level=info msg="Custom resource eclipse-che updated" 
time="2019-07-24T11:48:13Z" level=info msg="Updating eclipse-che CR with Eclipse Che server URL: http://che-che.192.168.99.114.nip.io " 
time="2019-07-24T11:48:13Z" level=info msg="Custom resource eclipse-che updated" 
time="2019-07-24T11:48:13Z" level=info msg="Eclipse Che is now available at: http://che-che.192.168.99.114.nip.io " 
time="2019-07-24T11:56:51Z" level=info msg="Updating eclipse-che CR with status: Che API: Unavailable" 
time="2019-07-24T11:56:51Z" level=info msg="Custom resource eclipse-che updated" 
time="2019-07-24T11:56:51Z" level=info msg="Waiting for deployment che. Default timeout: 420 seconds" 
time="2019-07-24T12:03:24Z" level=info msg="Deployment 'che' successfully scaled to 1" 
time="2019-07-24T12:03:24Z" level=info msg="Updating eclipse-che CR with Keycloak URL status: http://keycloak-che.192.168.99.114.nip.io " 
time="2019-07-24T12:03:24Z" level=info msg="Custom resource eclipse-che updated" 
time="2019-07-24T12:03:24Z" level=info msg="Updating eclipse-che CR with status: Eclipse Che server: Available" 
time="2019-07-24T12:03:24Z" level=info msg="Custom resource eclipse-che updated" 
time="2019-07-24T12:03:24Z" level=info msg="Updating eclipse-che CR with Eclipse Che server URL: http://che-che.192.168.99.114.nip.io " 
time="2019-07-24T12:03:24Z" level=info msg="Custom resource eclipse-che updated" 
time="2019-07-24T12:03:24Z" level=info msg="Eclipse Che is now available at: http://che-che.192.168.99.114.nip.io " 
```

**Comment by tsmaeder** [comment]
> Could you add logs of che-server pod ? out-of-memory ?
> 
> and could you list all kubernetes/openshift deployments objects

What do you mean by "list"? Contents?

**Comment by amisevsk** [comment]
Looks like your minishift cluster is evicting the Che pods, how much memory/disk are you giving minishift? What's the output of `minishift status`? 

I don't recall where events related to evictions would be found.

**Comment by sunix** [comment]
is the better without using the operator ( to "unblock" that issue)

**Comment by tsmaeder** [comment]
40Gig disk, 16G RAM

**Comment by amisevsk** [comment]
Only other thing I can think of is `oc describe node` as `system:admin` and checking the conditions section.

**Comment by benoitf** [comment]
@tsmaeder did you did a minishift start with fewer memory and disk and then start again with higher options or you deleted/started a new one

because If you just do stop/start with different settings AFAIK it is not applied on other start commands (but maybe now it's not as it was)

could you check
```
$ minishift config view
```

**Comment by nickboldt** [comment]
FWIW I don't encounter this when I deploy with the minishift addon instead of the operator:

```
minishift start
eval $(minishift docker-env)
oc adm policy  --as system:admin add-cluster-role-to-user cluster-admin developer
oc login -u developer -p 123

oc create secret generic registryredhatio --type=kubernetes.io/dockerconfigjson --from-file=.dockerconfigjson=${HOME}/.docker/config.json && oc secrets link default registryredhatio --for=pull && oc secrets link builder registryredhatio && oc get secret registryredhatio
```
then

chectl server:start -a minishift-addon -p minishift
# required if you're on minishift 1.34.0 or earlier
oc set env dc/che CHE_WORKSPACE_PLUGIN__REGISTRY__URL=https://che-plugin-registry.openshift.io/v3
```

**Comment by slemeur** [comment]
While this is a blocker to you @tsmaeder it seems that the solution is working on minikube. I would not make it a blocker as there is an existing path. But that is something to workout. Also please check the proposed solution from @nickboldt .

**Comment by mmorhun** [comment]
Tried to reproduce the issue but it worked for me.
What I did:
1. `minishift delete`
2. Download latest minishift (`minishift v1.34.1+c2ff9cb`).
3. Download latest chectl (`chectl/0.0.2-a74ad81 linux-x64 node-v10.4.1`)
4. `minishift start --vm-driver=virtualbox --cpus=4 --memory=16192 --disk-size=50GB --docker-opt userland-proxy=false`
5. `oc login -u system:admin`
6. `./chectl-linux server:start -p minishift -a operator` (after this I had Che multiuser deployed)
7. `./chectl-linux workspace:start -f https://raw.githubusercontent.com/eclipse/che-theia/master/devfiles/che-theia-all.devfile.yaml --access-token=<the-token>` (I retreived it from a dashboard query in browser)
8. In the workspace from a terminal in `che-dev` (imo, name is wrong, correct one should be `che-theia-dev`). Then from the terminal: `cd /projects/theia`, `che:theia init`, `yarn`. The build finished successfully.

On host I have Fedora 30 and VirtualBox 6.0.8

Screenshot from minishift dashboard:
![minishift-dashboard](https://user-images.githubusercontent.com/15607393/61856513-f47d7400-aeca-11e9-8ab1-94dcb28b5882.png)

P.S. From my experience with Minishift + Che I may say it is not that stable. It may work just fine today and later I may end up in spending a half of day trying to resolve some issues with it. So, I will not wonder if it doesn't work on your side right now.
If it is ok for you, give a try to Minikube + single user Che.
